### PR TITLE
E_DistributionMetadata DPU initial version

### PR DESCRIPTION
**Pull request by mkl-eea-sk**
Feature/e distribution metadata move

### CODE REVIEW

File: e-distributionMetadata/src/main/resources/build-info.properties [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Contains informations aboud DPU build. This file is used by
+# cz.cuni.mff.xrg.uv.boost.dpu.vaadin.AboutTab. Do not modify it's content!
+build.timestamp = ${build.timestamp}
+
+git.branch = ${git.branch}
+git.dirty = ${git.dirty}
+git.commit.id = ${git.commit.id}
+
+git.repository.link = ${git.repository.link}
```
Comment (tomas-knap) [inline e-distributionMetadata/src/main/resources/build-info.properties]: This should be removed as it points to CUNI repo I guess

File: e-distributionMetadata/pom.xml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<parent>
+        <groupId>cz.cuni.mff.xrg.uv</groupId>
+        <artifactId>uk-pom-dpu</artifactId>
```
Comment (tomas-knap) [inline e-distributionMetadata/pom.xml]: Parent proj should be changed to use the UV/Plugins' -uv-pom-dpu

File: e-distributionMetadata/pom.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        <version>1.0.0</version>
+	</parent>
+	<groupId>cz.opendata.unifiedviews.dpus</groupId>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: groupID = eu.unifiedviews.plugins

File: e-distributionMetadata/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
+# T-Metadata #
```
Comment (skrchnavy) [inline e-distributionMetadata/README.md]: wrong name of DPU

File: e-distributionMetadata/pom.xml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	<artifactId>E-DistributionMetadata</artifactId>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: artifactID shall be uv-e-distributionMetadata

File: e-distributionMetadata/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+----------
+
+###General###
+
+|                              |                                                               |
+|------------------------------|---------------------------------------------------------------|
+|**Name:**                     |T-Metadata                                              |
+|**Description:**              |Generates metadata on output from input. |
+|                              |                                                               |
+|**DPU class name:**           |Metadata     |
```
Comment (skrchnavy) [inline e-distributionMetadata/README.md]: is not aligned with pom.xml
<dpu.package>cz.opendata.unifiedviews.dpus.distributionMetadata</dpu.package>
<dpu.mainClass>DistributionMetadata</dpu.mainClass>

File: e-distributionMetadata/pom.xml [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+	<name>E-DistributionMetadata</name>
+	<description>Enables designers to input distribution level metadata</description>
+	<version>2.1.0</version>
+	<packaging>bundle</packaging>
+
+	<properties>
+		<dpu.package>cz.opendata.unifiedviews.dpus.distributionMetadata</dpu.package>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: move the package to eu.unifiedviews.plugins.transformer.distributionmetadata

File: e-distributionMetadata/pom.xml [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+		<dpu.mainClass>DistributionMetadata</dpu.mainClass>
+        <uv.helpers.version>[2.0.0-SNAPSHOT,3.0.0)</uv.helpers.version>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: reformat of pom.xml (tabs vs. spaces)

**Comment by skrchnavy** [comment]
Necessary to synchronize type of DPU, in README it is transformer, in other places it is extractor. It has input but optional (not clear why is there).

**Comment by mkl-eea-sk** [comment]
Hm,  I don't know, this is inherited from original DPU. So is it extractor or transformer? And what is the optional input. I simply do not know.

----- Original Message -----

**Comment by skrchnavy** [comment]
@tomas-knap is there a rule that defines difference between transformer and extractor? Is DPU with only  optional inputs T or E?

**Comment by tomas-knap** [comment]
Input is there because it may take certain metadata from dataset metadata being available as input. Since we decide to change the DPU (and do not try to use the DPU for ODN use case and at the same time for CUNI pipelines), we may remove such input. It is e-distributionMetadata, extractor.

**Comment by skrchnavy** [comment]
@mkl-eea-sk is it possible to merge or do you see there further changes?

File: e-distributionMetadata/pom.xml [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+	<artifactId>uv-e-distributionMetadata</artifactId>
+	<name>E-DistributionMetadata</name>
+	<description>Enables designers to input distribution level metadata</description>
+	<version>2.2.0-SNAPSHOT</version>
+	<packaging>bundle</packaging>
+
+	<properties>
+		<dpu.package>cz.opendata.unifiedviews.dpus.distributionMetadata</dpu.package>
+		<dpu.mainClass>DistributionMetadata</dpu.mainClass>
+		<uv.helpers.version>[2.0.0-SNAPSHOT,3.0.0)</uv.helpers.version>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: Is here range necessary? and is it correct range?

File: e-distributionMetadata/pom.xml [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+		<artifactId>uv-pom-dpu</artifactId>
+		<version>2.0.1</version>
+		<relativePath />
+	</parent>
+
+	<groupId>eu.unifiedviews.plugins</groupId>
+	<artifactId>uv-e-distributionMetadata</artifactId>
+	<name>E-DistributionMetadata</name>
+	<description>Enables designers to input distribution level metadata</description>
+	<version>2.2.0-SNAPSHOT</version>
```
Comment (skrchnavy) [inline e-distributionMetadata/pom.xml]: Is version 2.2.0 correct?

**Comment by skrchnavy** [comment]
Version and parent version shall be synchronized with #232

**Comment by mkl-eea-sk** [comment]
requirements changed during work, stop commenting not relevant things

**Pull request closed without merging**